src/chainstate/burn/db/sortdb.rs

```diff
@@ -2409,16 +2416,35 @@ impl SortitionDB {
         Ok(())
     }
 
-    fn check_schema_version_and_update(&mut self) -> Result<(), db_error> {
-        match self.get_schema_version() {
+    fn check_schema_version_and_error(&mut self) -> Result<(), db_error> {
+        match SortitionDB::get_schema_version(self.conn()) {
+            Ok(Some(version)) => {
+                let expected_version = SORTITION_DB_VERSION.to_string();
+                if version == expected_version {
+                    Ok(())
+                } else {
+                    Err(db_error::Other(format!(
+                        "The version of the sortition DB {} does not match the expected {} and cannot be updated from SortitionDB::open()",
+                        version, expected_version
+                    )))
+                }
+            }
+            Ok(None) => panic!("The schema version of the sortition DB is not recorded."),
+            Err(e) => panic!("Error obtaining the version of the sortition DB: {:?}", e),
+        }
+    }
+
+    fn check_schema_version_and_update(&mut self, epochs: &[StacksEpoch]) -> Result<(), db_error> {
+        let tx = self.tx_begin()?;
+        match SortitionDB::get_schema_version(&tx) {
             Ok(Some(version)) => {
                 let expected_version = SORTITION_DB_VERSION.to_string();
                 if version == expected_version {
                     return Ok(());
                 }
                 if version == "1" {
-                    let tx = self.tx_begin()?;
                     SortitionDB::apply_schema_2(&tx)?;
+                    SortitionDB::validate_and_insert_epochs(&tx, epochs)?;
```

Comment:
Should this be part of `apply_schema_2` since it's part of the migration code?

Reply:
Yep -- moved in 393757070